jsonschema2md/__init__.py

```diff
@@ -590,11 +610,72 @@ def _parse_object(
 
         return output_lines
 
+    def parse_file(
+        self,
+        file: Path,
+        fail_on_error_in_defs: bool = True,
+        locale: Optional[str] = None,
+    ) -> dict[str, Sequence[str]]:
+        """
+        Parse JSON Schema file and its references to Markdown text.
+
+        Parameters
+        ----------
+        file: Path
+            The Path to the JSON Schema file to parse.
+        fail_on_error_in_defs: bool
+            If True, the method will raise an error when encountering issues in the
+            "definitions" section of the schemas. If False, the method will attempt to continue parsing
+            despite such errors.
+        locale: Optional[str]
+            The locale to use for translations. If None, the default locale will be used.
+
+        Returns
+        -------
+        dict[str, Sequence[str]]
+            A dictionary where keys are file names (without `.json` extension) and values are lists of strings
+            representing the parsed Markdown documentation for each file.
+        """
+        if locale is not None:
+            Parser.current_locale = negotiate_locale((locale,), get_locales())
+
+        with file.open(encoding="utf-8") as input_file:
+            schema_obj = json.load(input_file)
+
+        root = self.parse_schema(schema_obj, fail_on_error_in_defs)
+        parsed_files = {"_js2md_root_": root}
```

Comment:
Maybe use the file name instead of _js2md_root_?
Even better would be to return Path objects.

Reply:
+1 :-)

Reply:
Shouldn't be like the other the json schema file name?

Reply:
Yeah, this is better

Reply:
Yeah this is what I meant, don't know why I used "out" as example lol